In [1]:
import json

import pandas as pd

In [2]:
import json
def read_comet(file_path):
    with open(file_path, 'r') as fh:
        for line in fh:
            vj = json.loads(line)
            yield {'story_id': vj['story_id'],
                   'obs1': vj['obs1'],
                   'obs2': vj['obs2'],
                   'real_hypothesis': vj['hyp1'] if vj['label'] == 1 else vj['hyp2'],
                   'comet_model': vj['generations']['gpt2_for_anli_comet'][0]}

In [3]:
def read_abduction(file_path):
    with open(file_path, 'r') as fh:
        next(fh)
        for line in fh:
            val = line.split(',')
            yield {'obs1': val[0],
                   'obs2': val[1],
                   'real_hypothesis': val[3],
                   'abduction_model': val[2]}

In [4]:
comet_df_validation = pd.DataFrame(read_comet('results/dev-w-comet-preds-generations.jsonl'))
comet_df_validation

story_id  \
0     0023fe98-5106-458d-8069-e72f1f7d65eb-1   
1     0023fe98-5106-458d-8069-e72f1f7d65eb-1   
2     0023fe98-5106-458d-8069-e72f1f7d65eb-1   
3     004b048d-beda-45a3-bb51-7a16dec21c5a-1   
4     004b048d-beda-45a3-bb51-7a16dec21c5a-1   
...                                      ...   
7247  ff76fad7-2c83-4173-9575-f627092a0846-1   
7248  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   
7249  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   
7250  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   
7251  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   

                                                   obs1  \
0                               Stephen was at a party.   
1                               Stephen was at a party.   
2                               Stephen was at a party.   
3     Donna keeps a giant bowl of candy on her desk ...   
4     Donna keeps a giant bowl of candy on her desk ...   
...                                                 ...   
7247                         My dog loved tennis balls.   
7248  Sandy wanted to know what her husband bought h...   
7249  Sandy wanted to know what her husband bought h...   
7250  Sandy wanted to know what her husband bought h...   
7251  Sandy wanted to know what her husband bought h...   

                                                   obs2  \
0           He checked it but it was completely broken.   
1           He checked it but it was completely broken.   
2           He checked it but it was completely broken.   
3     Her coworkers were so excited for all the new ...   
4     Her coworkers were so excited for all the new ...   
...                                                 ...   
7247  My dog had stolen every one and put it under t...   
7248           She carefully opened it and took a peek.   
7249           She carefully opened it and took a peek.   
7250           She carefully opened it and took a peek.   
7251           She carefully opened it and took a peek.   

                                        real_hypothesis  \
0              Stephen knocked over a vase while drunk.   
1                            Stephen dropped his phone.   
2     Stephen broke Jack's computer by accident whil...   
3                    Donna forgot to replace the candy.   
4                             Donna got some old candy.   
...                                                 ...   
7247                         My dog hates tennis balls.   
7248  Sandy accidentally stumbled upon her gift in t...   
7249            But she couldn't find where he hid hit.   
7250  When Sandy was at work, her husband searched t...   
7251  When he went to work the curiosity did not get...   

                                            comet_model  
0                            Stephen forgot his jacket.  
1                     Stephen fell asleep on the couch.  
2     Stephen was distracted from his father by a no...  
3                   Donna's co workers like spicy food.  
4                             Donna got some new boxes.  
...                                                 ...  
7247      My dog ran into a large ball when she saw it.  
7248                Sandy's husband made her something.  
7249                 Sandy decided to open a large box.  
7250                       Sandy wanted a real cupcake.  
7251  Sandy wanted to know what the tastes of the th...  

[7252 rows x 5 columns]

In [5]:
comet_df_test = pd.DataFrame(read_comet('results/test-w-comet-preds-generations.jsonl'))
comet_df_test

story_id  \
0      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
1      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
2      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
3      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
4      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
...                                       ...   
14308  fffa8585-8273-49ed-a818-460c981f9779-1   
14309  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   
14310  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   
14311  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   
14312  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   

                                                    obs1  \
0      The Smiths were having holidays done of the ch...   
1      The Smiths were having holidays done of the ch...   
2      The Smiths were having holidays done of the ch...   
3      The Smiths were having holidays done of the ch...   
4      The Smiths were having holidays done of the ch...   
...                                                  ...   
14308              Willy had butterflies in his stomach.   
14309                   My mom and I had a big argument.   
14310                   My mom and I had a big argument.   
14311                   My mom and I had a big argument.   
14312                   My mom and I had a big argument.   

                                                    obs2  \
0      Ty's face lit up as he ran to the new toy, hap...   
1      Ty's face lit up as he ran to the new toy, hap...   
2      Ty's face lit up as he ran to the new toy, hap...   
3      Ty's face lit up as he ran to the new toy, hap...   
4      Ty's face lit up as he ran to the new toy, hap...   
...                                                  ...   
14308                Willy played great and was the mvp.   
14309  I finally called my mom and the two of us reco...   
14310  I finally called my mom and the two of us reco...   
14311  I finally called my mom and the two of us reco...   
14312  I finally called my mom and the two of us reco...   

                                        real_hypothesis  \
0                  The Smiths bought toys for the kids.   
1                  The Smiths had not bought  presents.   
2                          The smiths forgot the gifts.   
3                They included some toys in the photos.   
4                    Their children came to their home.   
...                                                 ...   
14308            Willie stopped to relax for 5 minutes.   
14309  I didn't like that my mom and I weren't talking.   
14310    I refused to call her and we never reconciled.   
14311             My mom and I did not talk for awhile.   
14312                        We went our separate ways.   

                                             comet_model  
0      The Smiths bought a new toy, Ty had never seen...  
1      One of Ty's kids introduced himself, and Ty kn...  
2                                   Ty bought a new toy.  
3                            They bought a video camera.  
4                  Ty decided to get some birthday toys.  
...                                                  ...  
14308               Willy's favorite person is a Mr. It.  
14309                              I fell off the couch.  
14310                             My mom was devastated.  
14311                       My mom and I were separated.  
14312                  My mom was not too happy with us.  

[14313 rows x 5 columns]

In [6]:
abduction_model_gpt2_lg_validation = pd.DataFrame(read_abduction('results/results_gpt2_large_validation.csv'))
abduction_model_gpt2_lg_validation

obs1  \
0                               Stephen was at a party.   
1                               Stephen was at a party.   
2                               Stephen was at a party.   
3     Donna keeps a giant bowl of candy on her desk ...   
4     Donna keeps a giant bowl of candy on her desk ...   
...                                                 ...   
7247                         My dog loved tennis balls.   
7248  Sandy wanted to know what her husband bought h...   
7249  Sandy wanted to know what her husband bought h...   
7250  Sandy wanted to know what her husband bought h...   
7251  Sandy wanted to know what her husband bought h...   

                                                   obs2  \
0           He checked it but it was completely broken.   
1           He checked it but it was completely broken.   
2           He checked it but it was completely broken.   
3     Her coworkers were so excited for all the new ...   
4     Her coworkers were so excited for all the new ...   
...                                                 ...   
7247  My dog had stolen every one and put it under t...   
7248           She carefully opened it and took a peek.   
7249           She carefully opened it and took a peek.   
7250           She carefully opened it and took a peek.   
7251           She carefully opened it and took a peek.   

                                        real_hypothesis  \
0            Stephen knocked over a vase while drunk.\n   
1                          Stephen dropped his phone.\n   
2     Stephen broke Jack's computer by accident whil...   
3             Donna got new treats for her coworkers.\n   
4                           Donna got some new candy.\n   
...                                                 ...   
7247  One day I found all of my lost tennis balls un...   
7248              Sandy found her gift in the closet.\n   
7249       She found the present lying in plain site.\n   
7250  When her husband was at work Sandy searched th...   
7251  When he went to work the curiosity got the bes...   

                                        abduction_model  
0              Stephen dropped his phone.  was broken.   
1              Stephen dropped his phone.  was broken.   
2              Stephen dropped his phone.  was broken.   
3     Donna's coworkers gave her a box of candy.  ca...  
4     Donna's coworkers gave her a box of candy.  ca...  
...                                                 ...  
7247  I found several balls under my nightstand.  we...  
7248  When he went to work the curiosity got the bes...  
7249  When he went to work the curiosity got the bes...  
7250  When he went to work the curiosity got the bes...  
7251  When he went to work the curiosity got the bes...  

[7252 rows x 4 columns]

In [7]:
abduction_model_gpt2_lg_test = pd.DataFrame(read_abduction('results/results_gpt2_large_test.csv'))
abduction_model_gpt2_lg_test

obs1  \
0      The Smiths were having holidays done of the ch...   
1      The Smiths were having holidays done of the ch...   
2      The Smiths were having holidays done of the ch...   
3      The Smiths were having holidays done of the ch...   
4      The Smiths were having holidays done of the ch...   
...                                                  ...   
14308              Willy had butterflies in his stomach.   
14309                   My mom and I had a big argument.   
14310                   My mom and I had a big argument.   
14311                   My mom and I had a big argument.   
14312                   My mom and I had a big argument.   

                                                    obs2  \
0      Ty's face lit up as he ran to the new toy happ...   
1      Ty's face lit up as he ran to the new toy happ...   
2      Ty's face lit up as he ran to the new toy happ...   
3      Ty's face lit up as he ran to the new toy happ...   
4      Ty's face lit up as he ran to the new toy happ...   
...                                                  ...   
14308                Willy played great and was the mvp.   
14309  I finally called my mom and the two of us reco...   
14310  I finally called my mom and the two of us reco...   
14311  I finally called my mom and the two of us reco...   
14312  I finally called my mom and the two of us reco...   

                                         real_hypothesis  \
0                 The Smiths bought toys for the kids.\n   
1             The Smiths had bought a lot of presents.\n   
2                               The smiths gave gifts.\n   
3               They included some toys in the photos.\n   
4                   Their children came to their home.\n   
...                                                  ...   
14308           Willie stopped to relax for 5 minutes.\n   
14309  I didn't like that my mom and I weren't talkin...   
14310  We disagreed on how I was disciplining my chil...   
14311            My mom and I did not talk for awhile.\n   
14312                       We went our separate ways.\n   

                                         abduction_model  
0      The Smiths had bought a lot of presents.  was ...  
1      The Smiths had bought a lot of presents.  was ...  
2      The Smiths had bought a lot of presents.  was ...  
3      The Smiths had bought a lot of presents.  was ...  
4      The Smiths had bought a lot of presents.  was ...  
...                                                  ...  
14308  Willy was the starting quarterback of his team...  
14309  My mom and I did not talk for awhile.  was ter...  
14310  My mom and I did not talk for awhile.  was ter...  
14311  My mom and I did not talk for awhile.  was ter...  
14312  My mom and I did not talk for awhile.  was ter...  

[14313 rows x 4 columns]

In [8]:
abduction_model_gpt2_md_validation = pd.DataFrame(read_abduction('results/results_gpt2_medium_validation.csv'))
abduction_model_gpt2_md_validation

obs1  \
0                               Stephen was at a party.   
1                               Stephen was at a party.   
2                               Stephen was at a party.   
3     Donna keeps a giant bowl of candy on her desk ...   
4     Donna keeps a giant bowl of candy on her desk ...   
...                                                 ...   
7247                         My dog loved tennis balls.   
7248  Sandy wanted to know what her husband bought h...   
7249  Sandy wanted to know what her husband bought h...   
7250  Sandy wanted to know what her husband bought h...   
7251  Sandy wanted to know what her husband bought h...   

                                                   obs2  \
0           He checked it but it was completely broken.   
1           He checked it but it was completely broken.   
2           He checked it but it was completely broken.   
3     Her coworkers were so excited for all the new ...   
4     Her coworkers were so excited for all the new ...   
...                                                 ...   
7247  My dog had stolen every one and put it under t...   
7248           She carefully opened it and took a peek.   
7249           She carefully opened it and took a peek.   
7250           She carefully opened it and took a peek.   
7251           She carefully opened it and took a peek.   

                                        real_hypothesis  \
0            Stephen knocked over a vase while drunk.\n   
1                          Stephen dropped his phone.\n   
2     Stephen broke Jack's computer by accident whil...   
3             Donna got new treats for her coworkers.\n   
4                           Donna got some new candy.\n   
...                                                 ...   
7247  One day I found all of my lost tennis balls un...   
7248              Sandy found her gift in the closet.\n   
7249       She found the present lying in plain site.\n   
7250  When her husband was at work Sandy searched th...   
7251  When he went to work the curiosity got the bes...   

                                        abduction_model  
0                             Stephen broke his glass.   
1                             Stephen broke his glass.   
2                             Stephen broke his glass.   
3     Donna's coworkers were very excited for all th...  
4     Donna's coworkers were very excited for all th...  
...                                                 ...  
7247        I found several balls under my nightstand.   
7248  When he went to work the received a small box ...  
7249  When he went to work the received a small box ...  
7250  When he went to work the received a small box ...  
7251  When he went to work the received a small box ...  

[7252 rows x 4 columns]

In [9]:
abduction_model_gpt2_md_test = pd.DataFrame(read_abduction('results/results_gpt2_medium_test.csv'))
abduction_model_gpt2_md_test

obs1  \
0      The Smiths were having holidays done of the ch...   
1      The Smiths were having holidays done of the ch...   
2      The Smiths were having holidays done of the ch...   
3      The Smiths were having holidays done of the ch...   
4      The Smiths were having holidays done of the ch...   
...                                                  ...   
14308              Willy had butterflies in his stomach.   
14309                   My mom and I had a big argument.   
14310                   My mom and I had a big argument.   
14311                   My mom and I had a big argument.   
14312                   My mom and I had a big argument.   

                                                    obs2  \
0      Ty's face lit up as he ran to the new toy happ...   
1      Ty's face lit up as he ran to the new toy happ...   
2      Ty's face lit up as he ran to the new toy happ...   
3      Ty's face lit up as he ran to the new toy happ...   
4      Ty's face lit up as he ran to the new toy happ...   
...                                                  ...   
14308                Willy played great and was the mvp.   
14309  I finally called my mom and the two of us reco...   
14310  I finally called my mom and the two of us reco...   
14311  I finally called my mom and the two of us reco...   
14312  I finally called my mom and the two of us reco...   

                                         real_hypothesis  \
0                 The Smiths bought toys for the kids.\n   
1             The Smiths had bought a lot of presents.\n   
2                               The smiths gave gifts.\n   
3               They included some toys in the photos.\n   
4                   Their children came to their home.\n   
...                                                  ...   
14308           Willie stopped to relax for 5 minutes.\n   
14309  I didn't like that my mom and I weren't talkin...   
14310  We disagreed on how I was disciplining my chil...   
14311            My mom and I did not talk for awhile.\n   
14312                       We went our separate ways.\n   

                                         abduction_model  
0              The Smiths had bought a lot of presents.   
1              The Smiths had bought a lot of presents.   
2              The Smiths had bought a lot of presents.   
3              The Smiths had bought a lot of presents.   
4              The Smiths had bought a lot of presents.   
...                                                  ...  
14308   Willy was the starting quarterback of his team.   
14309  I didn't like that my mom and I weren't talking.   
14310  I didn't like that my mom and I weren't talking.   
14311  I didn't like that my mom and I weren't talking.   
14312  I didn't like that my mom and I weren't talking.   

[14313 rows x 4 columns]

In [10]:
abduction_model_gpt2_sm_validation = pd.DataFrame(read_abduction('results/results_gpt2_small_validation.csv'))
abduction_model_gpt2_sm_validation

obs1  \
0                               Stephen was at a party.   
1                               Stephen was at a party.   
2                               Stephen was at a party.   
3     Donna keeps a giant bowl of candy on her desk ...   
4     Donna keeps a giant bowl of candy on her desk ...   
...                                                 ...   
7247                         My dog loved tennis balls.   
7248  Sandy wanted to know what her husband bought h...   
7249  Sandy wanted to know what her husband bought h...   
7250  Sandy wanted to know what her husband bought h...   
7251  Sandy wanted to know what her husband bought h...   

                                                   obs2  \
0           He checked it but it was completely broken.   
1           He checked it but it was completely broken.   
2           He checked it but it was completely broken.   
3     Her coworkers were so excited for all the new ...   
4     Her coworkers were so excited for all the new ...   
...                                                 ...   
7247  My dog had stolen every one and put it under t...   
7248           She carefully opened it and took a peek.   
7249           She carefully opened it and took a peek.   
7250           She carefully opened it and took a peek.   
7251           She carefully opened it and took a peek.   

                                        real_hypothesis  \
0            Stephen knocked over a vase while drunk.\n   
1                          Stephen dropped his phone.\n   
2     Stephen broke Jack's computer by accident whil...   
3             Donna got new treats for her coworkers.\n   
4                           Donna got some new candy.\n   
...                                                 ...   
7247  One day I found all of my lost tennis balls un...   
7248              Sandy found her gift in the closet.\n   
7249       She found the present lying in plain site.\n   
7250  When her husband was at work Sandy searched th...   
7251  When he went to work the curiosity got the bes...   

                                        abduction_model  
0     Stephen was very drunk. Stephen drank too much...  
1     Stephen was very drunk. Stephen drank too much...  
2     Stephen was very drunk. Stephen drank too much...  
3     Donna's coworkers were so happy. Donna's cowor...  
4     Donna's coworkers were so happy. Donna's cowor...  
...                                                 ...  
7247  My dog ran away and I found it. [SEP]My dog ra...  
7248  Sandy was surprised to find a package. Sandy o...  
7249  Sandy was surprised to find a package. Sandy o...  
7250  Sandy was surprised to find a package. Sandy o...  
7251  Sandy was surprised to find a package. Sandy o...  

[7252 rows x 4 columns]

In [11]:
abduction_model_gpt2_sm_test = pd.DataFrame(read_abduction('results/results_gpt2_small_test.csv'))
abduction_model_gpt2_sm_test

obs1  \
0      The Smiths were having holidays done of the ch...   
1      The Smiths were having holidays done of the ch...   
2      The Smiths were having holidays done of the ch...   
3      The Smiths were having holidays done of the ch...   
4      The Smiths were having holidays done of the ch...   
...                                                  ...   
14308              Willy had butterflies in his stomach.   
14309                   My mom and I had a big argument.   
14310                   My mom and I had a big argument.   
14311                   My mom and I had a big argument.   
14312                   My mom and I had a big argument.   

                                                    obs2  \
0      Ty's face lit up as he ran to the new toy happ...   
1      Ty's face lit up as he ran to the new toy happ...   
2      Ty's face lit up as he ran to the new toy happ...   
3      Ty's face lit up as he ran to the new toy happ...   
4      Ty's face lit up as he ran to the new toy happ...   
...                                                  ...   
14308                Willy played great and was the mvp.   
14309  I finally called my mom and the two of us reco...   
14310  I finally called my mom and the two of us reco...   
14311  I finally called my mom and the two of us reco...   
14312  I finally called my mom and the two of us reco...   

                                         real_hypothesis  \
0                 The Smiths bought toys for the kids.\n   
1             The Smiths had bought a lot of presents.\n   
2                               The smiths gave gifts.\n   
3               They included some toys in the photos.\n   
4                   Their children came to their home.\n   
...                                                  ...   
14308           Willie stopped to relax for 5 minutes.\n   
14309  I didn't like that my mom and I weren't talkin...   
14310  We disagreed on how I was disciplining my chil...   
14311            My mom and I did not talk for awhile.\n   
14312                       We went our separate ways.\n   

                                         abduction_model  
0      Ty's parents were happy to see him. Ty's paren...  
1      Ty's parents were happy to see him. Ty's paren...  
2      Ty's parents were happy to see him. Ty's paren...  
3      Ty's parents were happy to see him. Ty's paren...  
4      Ty's parents were happy to see him. Ty's paren...  
...                                                  ...  
14308  Willy was the best mvp in the world. Willy was...  
14309  My mom told me she was mad at me. [SEP]I told ...  
14310  My mom told me she was mad at me. [SEP]I told ...  
14311  My mom told me she was mad at me. [SEP]I told ...  
14312  My mom told me she was mad at me. [SEP]I told ...  

[14313 rows x 4 columns]

In [12]:
comet_df_validation['abduction_gpt_lg_model'] = abduction_model_gpt2_lg_validation[['abduction_model']]
comet_df_validation['abduction_gpt_md_model'] = abduction_model_gpt2_md_validation[['abduction_model']]
comet_df_validation['abduction_gpt_sm_model'] = abduction_model_gpt2_sm_validation[['abduction_model']]
comet_df_validation

story_id  \
0     0023fe98-5106-458d-8069-e72f1f7d65eb-1   
1     0023fe98-5106-458d-8069-e72f1f7d65eb-1   
2     0023fe98-5106-458d-8069-e72f1f7d65eb-1   
3     004b048d-beda-45a3-bb51-7a16dec21c5a-1   
4     004b048d-beda-45a3-bb51-7a16dec21c5a-1   
...                                      ...   
7247  ff76fad7-2c83-4173-9575-f627092a0846-1   
7248  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   
7249  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   
7250  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   
7251  ffc2e6d7-723e-4dc2-ad33-e9757c4030db-1   

                                                   obs1  \
0                               Stephen was at a party.   
1                               Stephen was at a party.   
2                               Stephen was at a party.   
3     Donna keeps a giant bowl of candy on her desk ...   
4     Donna keeps a giant bowl of candy on her desk ...   
...                                                 ...   
7247                         My dog loved tennis balls.   
7248  Sandy wanted to know what her husband bought h...   
7249  Sandy wanted to know what her husband bought h...   
7250  Sandy wanted to know what her husband bought h...   
7251  Sandy wanted to know what her husband bought h...   

                                                   obs2  \
0           He checked it but it was completely broken.   
1           He checked it but it was completely broken.   
2           He checked it but it was completely broken.   
3     Her coworkers were so excited for all the new ...   
4     Her coworkers were so excited for all the new ...   
...                                                 ...   
7247  My dog had stolen every one and put it under t...   
7248           She carefully opened it and took a peek.   
7249           She carefully opened it and took a peek.   
7250           She carefully opened it and took a peek.   
7251           She carefully opened it and took a peek.   

                                        real_hypothesis  \
0              Stephen knocked over a vase while drunk.   
1                            Stephen dropped his phone.   
2     Stephen broke Jack's computer by accident whil...   
3                    Donna forgot to replace the candy.   
4                             Donna got some old candy.   
...                                                 ...   
7247                         My dog hates tennis balls.   
7248  Sandy accidentally stumbled upon her gift in t...   
7249            But she couldn't find where he hid hit.   
7250  When Sandy was at work, her husband searched t...   
7251  When he went to work the curiosity did not get...   

                                            comet_model  \
0                            Stephen forgot his jacket.   
1                     Stephen fell asleep on the couch.   
2     Stephen was distracted from his father by a no...   
3                   Donna's co workers like spicy food.   
4                             Donna got some new boxes.   
...                                                 ...   
7247      My dog ran into a large ball when she saw it.   
7248                Sandy's husband made her something.   
7249                 Sandy decided to open a large box.   
7250                       Sandy wanted a real cupcake.   
7251  Sandy wanted to know what the tastes of the th...   

                                 abduction_gpt_lg_model  \
0              Stephen dropped his phone.  was broken.    
1              Stephen dropped his phone.  was broken.    
2              Stephen dropped his phone.  was broken.    
3     Donna's coworkers gave her a box of candy.  ca...   
4     Donna's coworkers gave her a box of candy.  ca...   
...                                                 ...   
7247  I found several balls under my nightstand.  we...   
7248  When he went to work the curiosity got the bes...   
7249  When he went to work the curiosity got the bes...   
7250  When he went to work the cu

In [13]:
comet_df_test['abduction_gpt_lg_model'] = abduction_model_gpt2_lg_test[['abduction_model']]
comet_df_test['abduction_gpt_md_model'] = abduction_model_gpt2_md_test[['abduction_model']]
comet_df_test['abduction_gpt_sm_model'] = abduction_model_gpt2_sm_test[['abduction_model']]
comet_df_test

story_id  \
0      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
1      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
2      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
3      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
4      0061bad2-e595-483c-9649-87cc1f2ea3e8-1   
...                                       ...   
14308  fffa8585-8273-49ed-a818-460c981f9779-1   
14309  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   
14310  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   
14311  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   
14312  fffcd77b-26e2-4fa9-93b2-5dee55f9f452-1   

                                                    obs1  \
0      The Smiths were having holidays done of the ch...   
1      The Smiths were having holidays done of the ch...   
2      The Smiths were having holidays done of the ch...   
3      The Smiths were having holidays done of the ch...   
4      The Smiths were having holidays done of the ch...   
...                                                  ...   
14308              Willy had butterflies in his stomach.   
14309                   My mom and I had a big argument.   
14310                   My mom and I had a big argument.   
14311                   My mom and I had a big argument.   
14312                   My mom and I had a big argument.   

                                                    obs2  \
0      Ty's face lit up as he ran to the new toy, hap...   
1      Ty's face lit up as he ran to the new toy, hap...   
2      Ty's face lit up as he ran to the new toy, hap...   
3      Ty's face lit up as he ran to the new toy, hap...   
4      Ty's face lit up as he ran to the new toy, hap...   
...                                                  ...   
14308                Willy played great and was the mvp.   
14309  I finally called my mom and the two of us reco...   
14310  I finally called my mom and the two of us reco...   
14311  I finally called my mom and the two of us reco...   
14312  I finally called my mom and the two of us reco...   

                                        real_hypothesis  \
0                  The Smiths bought toys for the kids.   
1                  The Smiths had not bought  presents.   
2                          The smiths forgot the gifts.   
3                They included some toys in the photos.   
4                    Their children came to their home.   
...                                                 ...   
14308            Willie stopped to relax for 5 minutes.   
14309  I didn't like that my mom and I weren't talking.   
14310    I refused to call her and we never reconciled.   
14311             My mom and I did not talk for awhile.   
14312                        We went our separate ways.   

                                             comet_model  \
0      The Smiths bought a new toy, Ty had never seen...   
1      One of Ty's kids introduced himself, and Ty kn...   
2                                   Ty bought a new toy.   
3                            They bought a video camera.   
4                  Ty decided to get some birthday toys.   
...                                                  ...   
14308               Willy's favorite person is a Mr. It.   
14309                              I fell off the couch.   
14310                             My mom was devastated.   
14311                       My mom and I were separated.   
14312                  My mom was not too happy with us.   

                                  abduction_gpt_lg_model  \
0      The Smiths had bought a lot of presents.  was ...   
1      The Smiths had bought a lot of presents.  was ...   
2      The Smiths had bought a lot of presents.  was ...   
3      The Smiths had bought a lot of presents.  was ...   
4      The Smiths had bought a lot of presents.  was ...   
...                                                  ...   
14308  Willy was the starting quarterback of his team...   
14309  My mom and I did not talk for awhile.  was ter...   
14310  My mom and I did not talk for

In [14]:
comet_df_test.to_csv("./final_results/results_consolidated_testset.csv", index=False)
comet_df_validation.to_csv("./final_results/results_consolidated_evalset.csv", index=False)

#### Now to randomly select one row from each story id and from those we randomly select
#### 100 rows for human annotation

In [15]:
comet_df_test_100_sample = comet_df_test.groupby(by=['story_id'], axis=0).sample(n=1, random_state=10).sample(100, random_state=10)
comet_df_test_100_sample

story_id  \
9236   a4a142fa-0d92-4da9-b500-f8fdfbd65a69-1   
2460   2b648029-ba58-406b-bcc1-107fc27c0fc6-1   
13591  f315093b-1f5a-4009-8826-a08e4046231a-1   
6437   71e4828c-375d-4e9b-a156-04874875e0d3-1   
12998  e9897dcc-aa19-4ba6-a0f6-2325990a1b0a-1   
...                                       ...   
12258  dbc6ebba-14e7-4c1f-bdc4-8dc28ec27f4d-1   
3390   3b8d54a0-0d39-496a-bbae-86ffec8de983-1   
1821   1fdcff1a-d266-4abd-8291-386daccd2050-1   
4334   4d8a08d6-0950-4135-9889-45a8d75a4e30-1   
5986   6a88e010-b708-4e6e-a7a9-36cccd48fe77-1   

                                                    obs1  \
9236                   I got on the train to go to work.   
2460                                  James owned a cat.   
13591               I had to dissect a frog for biology.   
6437           Lucy was in the kitchen with her grandma.   
12998                     Neil was visiting Scandinavia.   
...                                                  ...   
12258  Laura has had pain while chewing food for months.   
3390                                   Brad's feet hurt.   
1821   Leon decided to go to a restaurant with a driv...   
4334                             What was cooking tacos.   
5986                 Dan was angry with his brother Rob.   

                                                    obs2  \
9236   Eventually the lights turned back on and we co...   
2460                 So he locked them out of the house.   
13591                         It was a great experience.   
6437   Lucy was shocked since her mother allowed her ...   
12998   Neil decided he would love to winter in Lapland!   
...                                                  ...   
12258  However, when the pain goes away, she is glad ...   
3390                     His feet felt better than ever!   
1821           They gave him his second burger for free.   
4334                        Whit burned the taco shells.   
5986         Dan didn't speak to his brother for a week.   

                                         real_hypothesis  \
9236                             No trains were working.   
2460   The cat posed for Instagram photos on James' e...   
13591  Seeing all the organs made the experience more...   
6437   But her non-gram did not let her use a steak k...   
12998                  Neil enjoyed his visit very much.   
...                                                  ...   
12258  Laura never went to the dentist and the orthod...   
3390        Brad burned his feet in a scolding hot bath.   
1821             The order taker got Leon's order wrong.   
4334   Whit left the taco shells in the fridge and go...   
5986    Rob had stolen some cash from Dan, enraging him.   

                                             comet_model  \
9236                                 It started raining.   
2460                James's cat ran outside in a ruckus.   
13591  I got to witness an animal-focused birth and m...   
6437                    Lucy tried to use a steak knife.   
12998                     It was always snowy in Europe.   
...                                                  ...   
12258                      Laura decided to have braces.   
3390   Brad decided to try running but needed to spra...   
1821                             Leon did his best work.   
4334                    Whit got all of his taco shells.   
5986          Dan could not decide whether to visit Rob.   

                                  abduction_gpt_lg_model  \
9236   The lights went out on the train.  the power w...   
2460                   The cat kept hat from his bed. .    
13591  I learned a lot from it.  was a my favorite ex...   
6437   Lucy wanted to use a steak knife to cut.  aske...   
12998  Neil went to visit three winter roads.  was th...   
...                                                  ...   
12258  Laura went to the dentist to have it removed. ...   
3390     Brad got new shoes.  came andBrad's feet sore.    
1821   The food arrived 

In [16]:
!pip install openpyxl
#fh = os.open('final_results/results_100sample.xlsx', 'w')
comet_df_test_100_sample.to_excel('final_results/results_100sample.xlsx',
                                  header=True, index=False)
#fh.close()

## From here on onwards we'll add evaluate human annotation results


In [84]:
score_ids = {1: 'human_score_of_abdcution_gpt_lg_model_as_compared_to_real_hypothesis',
             2: 'human_score_of_abdcution_gpt_md_model_as_compared_to_real_hypothesis',
             3: 'human_score_of_abdcution_gpt_sm_model_as_compared_to_real_hypothesis',
             4: 'human_score_of_abdcution_gpt_lg_model_as_compared_to_comet_model',
             5: 'human_score_of_abdcution_gpt_md_model_as_compared_to_comet_model',
             6: 'human_score_of_abdcution_gpt_sm_model_as_compared_to_comet_model'}
cols = list(score_ids.values()) + ['story_id']
human_ss = pd.read_excel('final_results/results_100sample_g2g_ss.xlsx', sheet_name='SunnyScoring')[cols]
human_kw = pd.read_excel('final_results/results_100sample_g2g_kw.xlsx')[cols]
human_tam = pd.read_excel('final_results/results_100sample_g2g_tam.xlsx')[cols]
human_alice = pd.read_excel('final_results/results_100sample_g2g_alice.xlsx')[cols]
human_chelsie = pd.read_excel('final_results/results_100sample_g2g_chelsie.xlsx')[cols]
annotators = {'ss': human_ss,
              'kw': human_kw,
              'tam': human_tam,
              'alice': human_alice,
              'chelsie': human_chelsie}

In [151]:
# human_score_of_abdcution_gpt_lg_model_as_compared_to_real_hypothesis
def extract_human_scores(annotators):
    dfs = []
    for annotator, annotated_df in annotators.items():
        print(f"Processing for {annotator}")
        df = pd.DataFrame()

        for k, v in score_ids.items():
            score_counts = annotated_df[[score_ids[k]]].value_counts()
            df[f"{k}_{3}"] = [score_counts[3]]
            df[f"{k}_{2}"] = [score_counts[2]]
            df[f"{k}_{1}"] = [score_counts[1]]

        print(f"finished creating dataframe for {annotator}")
        df.reindex([annotator])
        dfs.append(df)
    return pd.concat(dfs)


In [153]:
all_annotators_result = extract_human_scores(annotators)
all_annotators_result

Processing for ss
finished creating dataframe for ss
Processing for kw
finished creating dataframe for kw
Processing for tam
finished creating dataframe for tam
Processing for alice
finished creating dataframe for alice
Processing for chelsie
finished creating dataframe for chelsie


1_3  1_2  1_1  2_3  2_2  2_1  3_3  3_2  3_1  4_3  4_2  4_1  5_3  5_2  5_1  \
0   33   39   28   37   36   27   17   24   59   69   20   11   67   26    7   
0   63    3   34   52    4   43   46   25   29   58    6   36   52    6   41   
0   11   26   63   20   46   34   10   46   43   25   34   41   29   36   34   
0   12   12   76   44   36   20    1    8   91   15   10   75   53   27   20   
0   30   31   39   29   42   29   21   18   61   63   15   22   61   22   17   

   6_3  6_2  6_1  
0   27   57   16  
0   44   25   31  
0   14   50   36  
0    3    6   91  
0   41   23   36

In [154]:
all_annotators_result_mean = all_annotators_result.mean(axis=0)
all_annotators_result_mean

1_3    29.8
1_2    22.2
1_1    48.0
2_3    36.4
2_2    32.8
2_1    30.6
3_3    19.0
3_2    24.2
3_1    56.6
4_3    46.0
4_2    17.0
4_1    37.0
5_3    52.4
5_2    23.4
5_1    23.8
6_3    25.8
6_2    32.2
6_1    42.0
dtype: float64

## Human Judgments Result
| Model | less than neutral | neutral | greater than neutral | comparator
| --- | ----------- | -------| -----| ------|
| abd_gpt2_lg | 48.0 | 22.2 | 29.8 | real hypothesis |
| abd_gpt2_lg | 37 | 17 | 46 | comet |
| abd_gpt2_md | 30.6 | 32.8 | 36.4 | real hypothesis |
| abd_gpt2_md | 23.8 | 23.4 | 52.4 | comet |
| abd_gpt2_sm | 56.6 | 24.2 | 19.0 | real hypothesis |
| abd_gpt2_sm | 42.0 | 32.2 | 25.8 | comet |